WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt

```diff
@@ -389,8 +391,27 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
             val onCtaClick: ListItemInteraction,
         ) : Card(type = Type.WP_SOTW_2023_NUDGE_CARD)
 
-        data class NoCardsMessage(val title: UiString, val message: UiString)  : Card(Type.NO_CARDS_MESSAGE)
+        data class NoCardsMessage(val title: UiString, val message: UiString) : Card(Type.NO_CARDS_MESSAGE)
         data class PersonalizeCardModel(val onClick: () -> Unit) : Card(Type.PERSONALIZE_CARD)
+
+        data class Dynamic(
+            val id: String,
+            val order: Order,
+            val rows: List<Row>,
+            val title: String?,
+            val image: String?,
+            val action: String?,
+            val onHideMenuItemClick: ListItemInteraction,
+            val onCtaClick: ListItemInteraction,
```

Comment:
This is not a suggestion for a change, I'm just thinking out loud regarding the nullability of the `onCtaClick`.
I was thinking that:
* The `onCtaClick` should be non-null when we have an `action` string and a `url` from the model https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2923/files#diff-0834739cb5282253347464dba9388f8adc41facaa790a9fc28b0e5e64511ce1cR72
* When one of those is null there is no action and the action button should be hidden even if we have a string for it.

Wdyt?

Reply:
Totally makes sense. I revamped the feature according to this suggestion: p1702484824130069/1702473203.210019-slack-C067H36F011